### DISCUSSION TITLE
ViewOptionForm

### DISCUSSION THREAD

**Discussion by mikrosmile** [filamentphp/filament#3519] [Ideas]
Similar to createOptionForm in select, would be also great to have a viewForm with prefilled values based on selected option and link to specific model's resource.

**Comment by KaitoKid1612** [comment]
I also have the same situation

**Reply by KaitoKid1612** [reply]
![image](https://user-images.githubusercontent.com/43107595/235050018-be54f1d2-d197-4e77-a75b-b3f84529c55e.png)

**Reply by KaitoKid1612** [reply]
i want to click createOptionForm button it will redirect to detail page of this id:

Select::make('presentations_id')
                                    ->label('Presentation')
                                    ->options(Presentation::pluck('id', 'id')->toArray())
                                    ->required()
                                    ->reactive()
                                    ->afterStateUpdated(function ($state, callable $set) {
                                        $presentation = Presentation::find($state);
                                        $coAuthors = CoAuthor::where('presentation_id', $state)->get();
                                        $affiliations = $coAuthors->pluck('affiliation')->toArray();
                                        if ($presentation) {
                                            $presentation_format = PresentationFormat::find($presentation->presentation_format_id);
                                            $presentation_info = $presentation->title . PHP_EOL . implode(', ', $affiliations) . PHP_EOL . $presentation_format->name;
                                            $set('presentation_info', $presentation_info);
                                        }
                                    })
                                    ->afterStateHydrated(fn ($component) => $component->callAfterStateUpdated())
                                    ->createOptionForm(function ($get) {

                                    }),
Please help me do it

**Comment by lpd-au** [comment]
@danharrin thoughts?

**Reply by danharrin** [reply]
You can do this if you register your own Action on the field which reads the current field state and loads the modal content based on it
